app/dojo.py

```diff
@@ -0,0 +1,361 @@
+		#returns rooms with occupants
+		room_with_occupants = [room for room in self.all_rooms \
+		if room.occupants]
+
+		person_reallocate = [person for person \
+		in self.all_people if person.person_id == personID.upper()]
+
+		if not person_reallocate:
+			return "The person with id " + str(personID) + " doesn't exist"
+		#checks if person is in allocated a room
+		previous_room = [room for room in room_with_occupants\
+		 for occupant in room.occupants if occupant.upper() ==\
+		  person_reallocate[0].name]
+
+		if not previous_room:
+			return "person yet to be allocated a room"
+
+		#checks if room to rellocate to exists
+		new_room = [room for room in self.all_rooms \
+		if room.room_name.lower() == roomname.lower()]
+
+		if not new_room:
+			return "room " + roomname + " doesn't exists"
+		if new_room:
+			#checks if previous room type matches with new room type
+			roomtypes = [oldroom for oldroom in \
+			previous_room if oldroom.room_type == new_room[0].room_type]
+		if person_reallocate[0].category == "staff" \
+		and new_room[0].room_type == "living":
+			return "Can't rellocate staff to a living room"
+		if not roomtypes:
+			return "can't rellocate to a room of different type"
+		if new_room[0] in previous_room:
+			return "Can't reallocate to the same room"
+			#checks if room to rellocate to has space
+		if len(new_room[0].occupants) == new_room[0].max_capacity:
+			return "Sorry, room " + new_room[0].room_name.capitalize() + " is full"
+		else:
+			if new_room[0].room_type == previous_room[0].room_type:
+				new_room[0].occupants.append(person_reallocate[0].name)
+				previous_room[0].occupants.remove(person_reallocate[0].name)
+			else:
+				new_room[0].occupants.append(person_reallocate[0].name)
+				previous_room[1].occupants.remove(person_reallocate[0].name)
+			return person_reallocate[0].name + " " + "reallocated to " + roomname
+
+	def load_people(self, filename):
+		""" Loads people from a file to system """
+		response = ""
+		if not os.path.isfile(filename):
+			return "File {} doesn't exist".format(filename)
+		text_file = open(filename)
+		if os.path.getsize(filename) == 0:
+			response += "File {} is empty".format(filename)
+		for line in text_file:
+			person_details = line.rstrip().split()
+			if len(person_details) == 4 and person_details[2] \
+			in ("FELLOW", "STAFF") and person_details[3] in ("Y", "N"):
+				response += str(self.add_person(person_details[0], person_details[1], \
+				person_details[2], person_details[3])) + "\n\n"
+			elif len(person_details) == 3 and person_details[2] in \
+			("FELLOW", "STAFF"):
+				response += str(self.add_person(person_details[0], \
+				person_details[1], person_details[2])) + "\n\n"
+			else:
+				response += "Incorrect data format for -- {0}".format(line)
+		return response
+
+	def save_state(self, dbname="dojo"):
+		"""persists application data to database"""
+
+		engine = create_engine('sqlite:///database/{}.db'.format(dbname))
```

Comment:
what if the user wants to save state using a different name... you should check if there is a specified db name if not then set the db name to dojo.